meta-facebook/minerva-ag/src/platform/plat_i2c_target.c

```diff
@@ -27,17 +27,305 @@
 #include <zephyr.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
+#include "libutil.h"
 #include "plat_i2c_target.h"
+#include "plat_i2c.h"
+#include <logging/log.h>
+#include "plat_gpio.h"
+#include "plat_pldm_sensor.h"
+#include "pldm_sensor.h"
+#include "plat_version.h"
+#include "plat_isr.h"
 
+#define DEVICE_TYPE 0x01
+#define REGISTER_LAYOUT_VERSION 0x01
+#define AEGIS_CARRIER_BOARD_ID 0x0000
+#define AEGIS_CPLD_ADDR (0x4C >> 1)
+
+typedef struct __attribute__((__packed__)) {
+	uint8_t device_type;
+	uint8_t register_layout_version;
+	uint8_t num_idx; //Number of PDR indexes in this register
+	uint8_t reserved_1;
+
+	uint16_t sbi; //Sensor base index (SBI= 248*R) used in this register's array
+	uint16_t max_pdr_idx; //Max PDR sensor index exported in total (MAX_PDRIDX)
+
+	uint8_t sensor_r_len[]; //sensor[0] reading length ~ sensor[247] reading length
+} plat_sensor_init_data_0_1;
+// size = sizeof(plat_sensor_init_data_0_1) + num_idx
+
+typedef struct __attribute__((__packed__)) {
+	uint8_t sensor_index_offset; // Sensor index offset (e.g. PDR sensor index offset)
+	uint32_t sensor_value; // Sensor value (4 bytes)
+} sensor_entry;
+
+typedef struct __attribute__((__packed__)) {
+	uint8_t device_type; // Device type (Aegis = 0x01, Rainbow = 0x02)
+	uint8_t register_layout_version; // Register layout version (e.g. VERSION_1 = 0x01)
+	uint16_t sensor_base_index; // Sensor base index (SBI)
+	uint8_t max_sbi_off; // Max sensor base index offset in this register (0 <= MAX_SBI_OFF <= 49)
+	// The following is a flexible array of sensor entries.
+	// The number of entries is (max_sbi_off + 1)
+	sensor_entry sensor_entries[];
+} plat_sensor_init_data_0_2;
+// size = sizeof(plat_sensor_init_data_0_2) + num_sensors * sizeof(SensorEntry);
+// num_sensors = max_sbi_off + 1
+
+typedef struct __attribute__((__packed__)) {
+	uint16_t carrier_board_id; //  MTIA Gen1 - Aegis=0x00
+	uint32_t bic_fw_version;
+	uint32_t cpld_fw_version;
+
+	// FRU section
+} plat_sensor_init_data_0_6;
+// size = sizeof(plat_sensor_init_data_0_6)
+
+LOG_MODULE_REGISTER(plat_i2c_target);
 /* I2C target init-enable table */
 const bool I2C_TARGET_ENABLE_TABLE[MAX_TARGET_NUM] = {
 	TARGET_DISABLE, TARGET_DISABLE, TARGET_DISABLE, TARGET_ENABLE,
-	TARGET_DISABLE, TARGET_ENABLE,	TARGET_DISABLE, TARGET_DISABLE,
+	TARGET_DISABLE, TARGET_DISABLE, TARGET_ENABLE,	TARGET_DISABLE,
 	TARGET_DISABLE, TARGET_DISABLE, TARGET_DISABLE, TARGET_DISABLE,
 };
 
+plat_sensor_init_data_0_1 *sensor_init_data_0_1_table[2] = { NULL };
+plat_sensor_init_data_0_2 *sensor_init_data_0_2_table[4] = { NULL };
+plat_sensor_init_data_0_6 *sensor_init_data_0_6_table[1] = { NULL };
+
+void init_sensor_data_0_1_table()
+{
+	for (int reg_offset = 0; reg_offset < 2; reg_offset++) {
```

Comment:
For all of these `for` loops, that use reg_offset, it's not clear that you are going through all the elements in the corresponding array. Can we create variables/constants to make the relationship clear?
```
#define DATA_0_1_TABLE_LENGTH 2
plat_sensor_init_data_0_1 *sensor_init_data_0_1_table[DATA_0_1_TABLE_LENGTH] = ...
for (int reg_offset = 0; reg_offset < DATA_0_1_TABLE_LENGTH; reg_offset++) {
```